### ISSUE TITLE
Android TalkBack doesn't work on right screen on dual screen smartphone

### ISSUE THREAD

**Issue by lookuper** [flutter/flutter#94587]
## Steps to Reproduce

Reproducible on any flutter app. 

1. Open flutter app on dual screen device (Surface Duo 1/2 for example)
2. Move the app to the right screen
3. Enable Android TalkBack (screen narrator)
4. Try to click/move focues to any item in the app.

When the app open on left the screen, or in dual screen mode, TalkBack works as expected.

**Comment by maheshmnj** [comment]
Hi @lookuper, Thanks for filing the issue. Please share your output of `flutter doctor -v` on which the app was built.

**Comment by lookuper** [comment]
> Hi @lookuper, Thanks for filing the issue. Please share your output of `flutter doctor -v` on which the app was built.

Added. Also any Android app supports left/right/dual screen TalkBack function as expected.

**Comment by andreidiaconu** [comment]
Writing a few of my findings here. I have found two separate bugs while investigating this:

Steps to reproduce:
  - Make sure TalkBack is inactive.
  - Start the app on the right screen (hello world from `flutter create .` should do)
  - Start TalkBack using a hardware shortcut (to ensure no UI refresh is triggered by touch)
  - Use a gesture shortcut to make TalkBack dump what it sees to logs

I have seen two different behaviours unfold:

1. Node positions are out of the bounds of the Activity, making them invisible to TalkBack
    - Some node bounds are reported on the **left** screen (offset is 0,0) and because the root node and the activity are reported on the *right* screen (offset is 1410, 0), TalkBack considers them invisible.
    - Note that sometimes some of the nodes are reported correctly. The first node considered a "View" is always reported wrong, and because that is the root of everything, they are all considered invisible even when the numbers seem right for an individual node.
    - TalkBack also complains that there are focusable nodes that are not visible (`Focus, is focusable and has no visible children: prints node information`)
    - I suspect that this bug can be reproduced on tablets or other devices with multi-window support.
<img width="888" alt="image" src="https://user-images.githubusercontent.com/1402046/198692976-2a92afbe-81b3-4395-8d0a-6299030515ab.png">

2. The Accessibility Bridge has an empty Semantics tree.
    - The logs from TalkBack only have a root view (corresponding to FlutterView) with no children.
    - The Flutter semantics tree is communicated to the Flutter accessibility bridge *only* [if accessibility is enabled](https://github.com/flutter/engine/blob/8bb0441e592f104bbd0139d1b813020f6627b722/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L358). This is by design.
    - From debugging, I noticed that the [semantics map cached inside the accessibility bridge](https://github.com/flutter/engine/blob/8bb0441e592f104bbd0139d1b813020f6627b722/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L165) is *empty* in some situations when it should be populated.
    - Seems like enabling accessibility does not also trigger a Semantics tree update in certain situations. I have yet to reliably reproduce the behavior.
    - This version of the bug should be reproducible on any Android device, although I did only test on a Surface Duo.

Regardless of the exact way it fails, using the steps to reproduce on a Surface Duo always ends up providing no accessibility for users.
For behavior no 1, at this point I am not sure if there is something unusual happening on the Surface Duo or if this can be reproduced using a tablet with multi-window support. Testing on a tablet should shed light on that.
For behavior no 2, it looks like a race condition in the way that accessibility is both enabled in the bridge and the semantics is populated and should be reproducible on any Android device.

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?
